### PR TITLE
Improve usability on README.md

### PR DESCRIPTION
**Pull request by chaisupt**
In this pull request, I have made updates to the README file to improve its usability based on a cognitive walkthrough test. The changes aim to enhance clarity, organization, and user-friendliness. Based on my cognitive walkthrough test, there are several point on the README that I can improve. These improvements aim to provide a more intuitive and accessible experience for users who are new to the project. By optimizing the README, we can help users understand the project's purpose, get started quickly, and find relevant resources. I would appreciate a review of the changes made to the README. Please let me know if the modifications effectively address the usability issues identified in the cognitive walkthrough test. Any additional suggestions for further improvement are welcome.

In addition, here is my Cognitive walkthrough file and Test/Inspection Report
[chaisupt_spotipy_Cognitive Walkthrough.pdf](https://github.com/spotipy-dev/spotipy/files/11645312/chaisupt_spotipy_Cognitive.Walkthrough.pdf)
[chaisupt_Test_Inspection Report.pdf](https://github.com/spotipy-dev/spotipy/files/11645313/chaisupt_Test_Inspection.Report.pdf)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,3 +108,17 @@ Don’t forget to add the *Spotipy* tag, and any other relevant tags as well, be
 If you have suggestions, bugs or other issues specific to this library,
 file them [here](https://github.com/plamere/spotipy/issues).
 Or just send a pull request.
+
+## Contributing
+
+If you are a developer with Python experience, and you would like to contribute to Spotipy, please be sure to follow the guidelines listed on documentation page
+
+> #### [Visit the guideline](https://spotipy.readthedocs.io/en/2.22.1/#contribute)
```
Comment (hugovk) [inline README.md:116]: Don't link to a specific version, it will go out of date. Instead use the `latest` or `master` versions.

File: README.md [hunk 2]
```diff
@@ -77,3 +108,17 @@ Don’t forget to add the *Spotipy* tag, and any other relevant tags as well, be
...
+
+## License
```
Comment (hugovk) [inline README.md:118]: I'm not sure if this section is needed, as GitHub has very recently added tabs at the top to link to the licence among others:

image

File: README.md [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+- [License](#license)
+
+## Features
+
+Spotipy supports all of the features of the Spotify Web API including access to all end points, and support for user authorization. For details on the capabilities you are encouraged to review the [Spotify Web API](https://developer.spotify.com/web-api/) documentation.
+
 ## Documentation
 
-Spotipy's full documentation is online at [Spotipy Documentation](http://spotipy.readthedocs.org/).
+Spotipy's full documentation is online at [Spotipy Documentation](http://spotipy.readthedocs.org/). Some function may need a specific scope. Please visit https://developer.spotify.com/documentation/web-api/concepts/scopes to learn more. If you do not define the scope properly `ERROR 401 Unauthorized, permission missing` may occur.
```
Comment (hugovk) [inline README.md:23]:
```suggestion
Spotipy's [full documentation is online](http://spotipy.readthedocs.org/). Some function may need a [specific scope](https://developer.spotify.com/documentation/web-api/concepts/scopes). If you do not define the scope properly `ERROR 401 Unauthorized, permission missing` may occur.
```

File: README.md [hunk 4]
```diff
@@ -30,10 +44,9 @@ pip install spotipy --upgrade
 
 A full set of examples can be found in the [online documentation](http://spotipy.readthedocs.org/) and in the [Spotipy examples directory](https://github.com/plamere/spotipy/tree/master/examples).
 
-To get started, install spotipy and create an app on https://developers.spotify.com/.
-Add your new ID and SECRET to your environment:
+To get started, [install spotipy](#installation), create a new account or login on https://developers.spotify.com/. Go to the [dashboard](https://developer.spotify.com/dashboard), create an app and add your new ID and SECRET (ID and SECRET can be found on an app setting) to your environment ([step-by-step video](https://www.youtube.com/watch?v=kaBVN8uP358&t=0s)):
```
Comment (hugovk) [inline README.md:47]:
```suggestion
To get started, [install spotipy](#installation), create a new account or log in on https://developers.spotify.com/. Go to the [dashboard](https://developer.spotify.com/dashboard), create an app and add your new ID and SECRET (ID and SECRET can be found on an app setting) to your environment ([step-by-step video](https://www.youtube.com/watch?v=kaBVN8uP358)):
```

File: README.md [hunk 5]
```diff
@@ -30,10 +44,9 @@ pip install spotipy --upgrade
...
 
-### Without user authentication
+### Without user authentication Example
```
Comment (hugovk) [inline README.md:49]:
```suggestion
### Example without user authentication
```

File: README.md [hunk 6]
```diff
@@ -46,8 +59,20 @@ results = sp.search(q='weezer', limit=20)
...
+.
+.
+.
+18 Troublemaker
+19 Feels Like Summer
+```
+
 
-### With user authentication
+### With user authentication Example
```
Comment (hugovk) [inline README.md:75]:
```suggestion
### Example with user authentication
```

**Review by stephanebruckert — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -18,6 +18,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Changes the YouTube video link for authentication tutorial (the old video was in low definition, the new one is in high definition)
 - Updated links to Spotify in documentation 
 
+## [2.23.1] - 2023-12-28
```
Comment (stephanebruckert) [inline CHANGELOG.md:21]: 2.23.1 isn't released yet, can you merge this section with `Unreleased` above?

File: CHANGELOG.md [hunk 8]
```diff
@@ -18,6 +18,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Added
+- Added Table of Contents, Features, Contributing, and License section on README file
+- Added Expected result for quick start section
+
+### Changed
+- Make the short description clearer
+- Add some notice on Documentation section
+- Edit the Installation description to be clearer with step-by-step video link
```
Comment (stephanebruckert) [inline CHANGELOG.md:30]: no need to be so specific, you can just add:

> Improve usability on README.md

in `Changed`

File: CHANGELOG.md (hunk 7 above)
Comment (chaisupt) [reply CHANGELOG.md]: Thanks for your advice. I moved it to `Unreleased` as you advised already.

**Pull request merged**